pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1007,79 +1007,96 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
 			}
 		}
 
-	case networking.ClientTLSSettings_SIMPLE:
-		res := model.SdsCertificateConfig{
-			CaCertificatePath: model.GetOrDefault(proxy.Metadata.TLSClientRootCert, tls.CaCertificates),
-		}
+	case networking.ClientTLSSettings_SIMPLE, networking.ClientTLSSettings_MUTUAL:
```

Comment:
this function is massive and hard to follow (before and after your PR). Its confusing to have the switch cover two case (simple and mutual) yet then inside the switch branch on these two.

Can we break this up a bit? Rather than major nesting, we can exit early. For example, if tls.credentialName is set, exit early instead of going into another `else{}` branch

Reply:
Did some code duplication for CredentialName but it should improve readability now